chia/wallet/wallet_blockchain.py

```diff
@@ -186,7 +186,7 @@ async def get_peak_block(self) -> Optional[HeaderBlock]:
 
     async def set_finished_sync_up_to(self, height: int):
         if height > await self.get_finished_sync_up_to():
-            await self._basic_store.set_object("FINISHED_SYNC_UP_TO", uint32(height))
+            await self._basic_store.set_object("FINISHED_SYNC_UP_TO", uint32(height), True)
```

Comment:
This is not always True, there are places where we call `set_finished_sync_up_to` outside of a transaction